src/libPMacc/include/random/distributions/uniform/Uniform_float.hpp

```diff
@@ -48,10 +54,125 @@ namespace detail
         DINLINE result_type
         operator()(StateType& state)
         {
-            return _curand_uniform(RNGMethod().get32Bits(state));
+            const result_type 2pow32Inv = 2.3283064e-10f;
+            const uint32_t random = RNGMethod().get32Bits(state);
+            return static_cast<result_type>( random ) * 2pow32Inv +
+                ( 2pow32Inv / 2.0f );
         }
     };
 
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * Swap the value one to zero (creates a small error in uniform distribution)
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne<float>::SwapOneToZero,
+        T_RNGMethod,
+        void
+    >
+    {
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+    public:
+        typedef float result_type;
+
+        DINLINE result_type
+        operator()(StateType& state)
+        {
+            const result_type randomValue =
+                PMacc::random::distributions::Uniform<
+                    uniform::ExcludeZero<result_type>
+            >()(state);
+            return randomValue == 1.0f ? 0.0f : randomValue;
+        }
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * Reduce the random range to 2^24.
+     * Creates intervals with the width of epsilon/2.
```

Comment:
no that's fine - this reduces the random range of the uniform integer random number to a random value in [0, 2^24-1].